test/autobot.test.ts

```diff
@@ -0,0 +1,52 @@
+import { Autobot, PRContext } from "../src/autobot";
+import { Application } from "probot";
+
+const app = (Symbol("app") as unknown) as Application;
+const context = ({} as unknown) as PRContext;
+
+describe("autobot", () => {
+  describe("start", () => {
+    it("creates a new autobot instance", () => {
+      const autobot = Autobot.start(app, []);
+      expect(autobot).toBeInstanceOf(Autobot);
+    });
+
+    it("calls the onStart hook with the correct arguments", () => {
+      const autobot = Autobot.start(app, []);
+      // @ts-ignore
+      expect(autobot.hooks.app.onStart.call).toBeCalledWith(app);
+
+      // Ensures the default mock impl of call from tapable was called w/ the correct args
+      expect.assertions(2);
+    });
+  });
+
+  describe("onPullRequestReceived", () => {
+    const autobot = Autobot.start(app, []);
+    // @ts-ignore
+    autobot.getConfig = jest.fn().mockImplementation(() => Promise.resolve({}));
+    jest.fn().mockResolvedValueOnce;
+
+    type ConfigResult = Promise<never | object>;
+    const withConfig = (fnOrObj: (() => ConfigResult) | any) => {
+      if (typeof fnOrObj === "function") {
+        // @ts-ignore
+        (autobot.getConfig as jest.Mock<any, any>).mockImplementationOnce(fnOrObj);
+      } else if (fnOrObj instanceof Error) {
+        // @ts-ignore
+        (autobot.getConfig as jest.Mock<any, any>).mockImplementationOnce(() => Promise.reject(fnOrObj));
```

Comment:
I had to use these `mockImplementation` methods instead of `rejectWithValue` or w/e it is that jest provides because it turns out that's not a real promise and `await-to-js` chokes... who knew?